projects/storefrontlib/src/shared/services/promotion/promotion.service.ts

```diff
@@ -1,119 +1,29 @@
-        return this.getOrderPromotionsFromCheckout();
-      case PromotionLocation.Order:
-        return this.getOrderPromotionsFromOrder();
-      default:
-        return of([]);
-    }
-  }
-
-  getOrderPromotionsFromCart(): Observable<PromotionResult[]> {
-    return this.activeCartService
-      .getActive()
-      .pipe(map((cart) => this.getOrderPromotionsFromCartHelper(cart)));
-  }
-
-  private getOrderPromotionsFromCartHelper(cart: Cart): PromotionResult[] {
-    const potentialPromotions = [];
-    potentialPromotions.push(...(cart.potentialOrderPromotions || []));
-
-    const appliedPromotions = [];
-    appliedPromotions.push(...(cart.appliedOrderPromotions || []));
-
-    return [...potentialPromotions, ...appliedPromotions];
-  }
-
-  getOrderPromotionsFromCheckout(): Observable<PromotionResult[]> {
-    return this.checkoutService
-      .getOrderDetails()
-      .pipe(map((order) => this.getOrderPromotionsFromOrderHelper(order)));
-  }
-
-  getOrderPromotionsFromOrder(): Observable<PromotionResult[]> {
-    return this.orderDetailsService
-      .getOrderDetails()
-      .pipe(map((order) => this.getOrderPromotionsFromOrderHelper(order)));
-  }
-
-  private getOrderPromotionsFromOrderHelper(order: Order): PromotionResult[] {
-    const appliedOrderPromotions = [];
-    appliedOrderPromotions.push(...(order.appliedOrderPromotions || []));
-
-    return appliedOrderPromotions;
-  }
-
-  getProductPromotionForEntry(
-    item: OrderEntry,
-    promotionLocation: PromotionLocation
-  ): Observable<PromotionResult[]> {
-    switch (promotionLocation) {
-      case PromotionLocation.ActiveCart:
-        return this.activeCartService
-          .getActive()
-          .pipe(
-            map((cart) =>
-              this.getProductPromotion(
-                item,
-                cart.appliedProductPromotions || []
-              )
-            )
-          );
-      case PromotionLocation.Checkout:
-        return this.checkoutService
-          .getOrderDetails()
-          .pipe(
-            map((order) =>
-              this.getProductPromotion(
-                item,
-                order.appliedProductPromotions || []
-              )
-            )
-          );
-      case PromotionLocation.Order:
-        return this.orderDetailsService
-          .getOrderDetails()
-          .pipe(
-            map((order) =>
-              this.getProductPromotion(
-                item,
-                order.appliedProductPromotions || []
-              )
-            )
-          );
-    }
-  }
-
-  private getProductPromotion(
+import { Cart, Order, OrderEntry, PromotionResult } from '@spartacus/core';
+import { Observable } from 'rxjs';
+
+export abstract class PromotionService {
+  abstract getOrderPromotions(): Observable<PromotionResult[]>;
+
+  abstract getProductPromotionForEntry(
+    item: OrderEntry
+  ): Observable<PromotionResult[]>;
+
+  getProductPromotionForAllEntries(
+    order: Order | Cart
```

Comment:
I think, we can pass `entries: OrderEntry[]` directly, instead of ` order: Order | Cart`.  Since, `cart-item-list.component` has `items` as input, then we only need to call this function in `cart-item-list.component`.